File: frontend/src/services/workforceService.js
import api from './api';

const workforceService = {
  async list() {
    const response = await api.get('/workforce');
    return response.data?.data || [];
  },
  async get(userId) {
    const response = await api.get(`/workforce/${userId}`);
    return response.data?.data;
  },
  async updateStatus(userId, enabled) {
    const response = await api.patch(`/workforce/${userId}/status`, { enabled });
    return response.data?.data;
  },
  async remove(userId) {
    const response = await api.delete(`/workforce/${userId}`);
    return response.data?.data;
  },
};

export default workforceService;

File: backend/src/main/java/com/buildtrack/ai/service/WorkforceService.java
package com.buildtrack.ai.service;

import com.buildtrack.ai.auth.entity.User;
import com.buildtrack.ai.dto.workforce.WorkforceMemberResponse;

import java.util.List;

public interface WorkforceService {
    List<WorkforceMemberResponse> getAccessibleWorkforce(User user);
    WorkforceMemberResponse getMember(Long userId, User actor);
    WorkforceMemberResponse updateStatus(Long userId, boolean enabled, User actor);
    void removeMember(Long userId, User actor);
}
